proposal.rst

```diff
@@ -267,29 +285,29 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-* Weeks of bonding: get familiar with ranking algorithms, search for suitable datasets, assessment of the complexity of integrating large databases, and performance tests.
 
-In each next position assumes that testing and writing cross-platform part is included.
+* Weeks of bonding: Get more familiar with ranking algorithms, select specific suitable parts of the above datasets, discuss what is wrong with `#278 <https://github.com/xapian/xapian/pull/278>`_.
 
-* 1 Week - integrate datasets 
+* 1 Week - FIRE dataset integration. Each user have to download it by themself, because of 
+registration required. Implement training process using Letor API.  
 
-* 2 Weeks of exams (situation with exams is very unstable now in my University, so I timeline may change in this case)
+* 2 Weeks of exams (situation with exams is very unstable now in my University, so I timeline may change in this case).
 
-* 1 Week - solve possible problems with the size of datasets
+* 1 Week - add function for selecting scorer and ranker. It would be nice to write API allows user to select default sets of queries to run and run specified queries. For that purpose update build system to be able to run suite from terminal.
 
-* 3 Weeks - implement ListMLE, ListNet and SVM evaluation api
+* 1 Week - TREC dataset integration. It has similar format to FIRE, it should not be problem to integrate it. Start writing tool for combine docs.
 
-If no troubles appeared, I can try to develop suite for performance testing:
+* 1 Week - combine tool API should support plain logical operations, it could take time to implement them.
 
-* 1 Week - get familiar with data indexing and searching api
+* 1 Week - prepare tool for text data extracting from Wikipedia.
 
-* 1.5 Weeks - tests for indexing
+* 1 Week - add functionallity to perftest to drop system cache for more accurate results and run one case multiple times to get average time consuming.
```

Comment:
What do you mean by "system cache"? Are you talking about the VM / filesystem cache that most Unix-like operating systems have? How are you going to drop that? Is a run with a cold cache indicative of real-life use? Are you talking about running multiple times dropping the cache each time, or dropping the cache at the beginning then running a number of times? How many times is sensible? Should that be user configured?